internal/expr/parser.go

```diff
@@ -282,7 +303,11 @@ func (p *Parser) skipBlanks() bool {
 			continue
 		}
 		switch p.input[p.pos] {
-		case ' ', '\t', '\r', '\n':
+		case '\n':
+			p.firstByteOfLine = p.pos + 1
+			p.lineNum++
```

Comment:
Integrating this check into a function to replace `p.pos++` would mean you didn't have to copy this code for the third time.

Reply:
I have thought about this and about your above comment, the problem is that we are not always using `p.pos` as there are other instances in the code like `advance` where we check the *previous* character instead. We can probably change the parser but that will involve a more general refactor which I am not sure belongs to this PR.

Reply:
I was thinking that you could even use this new function instead of the p.pos++ in `advance`. There is only one other case I can think of and that is the fold in skipString (which we use because it is case insensitive), but honestly I think it would be fine to just not worry about that. You could put a comment above that function noting that it doesn't handle line breaks but I think it would also be very strange if we added code to this library that hit that case so its not necessary really.

Are there any other cases you were worried about or other concerns for this? I think if not then this PR is as good a place as any to include it, rather than doing a full refactor.

Reply:
I refactored the code, the only place left is `skipStringLiteral` which I will finish soon, I am thinking about the edge cases.

Reply:
Done as well, I had to change one comment which was a little bit misleading but the rest was straightforward.